src/libcore/mem.rs

```diff
@@ -611,3 +611,35 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
+/// representation is. Note that if the enum's representation is signed, then
+/// the discriminant values returned by this function will not compare in the
+/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
+/// 
+/// If `T` is not an enum, the return value is unspecified.
```

Comment:
It may also be worth explaining that the returned value only really makes sense to compare with other discriminants generated from values of the same type? That is, comparing the discriminants of types `A` and `B` won't result in anything meaningful